File: 033-Search-in-Rotated-Sorted-Array/Python/Solution02.py
#!/usr/bin/python
# ==============================================================================
# Author:   Tao Li (taoli@ucsd.edu)
# Date:     Jul 9, 2015
# Question: 033-Search-in-Rotated-Sorted-Array
# Link:     https://leetcode.com/problems/search-in-rotated-sorted-array/
# ==============================================================================
# Suppose a sorted array is rotated at some pivot unknown to you beforehand.
# 
# (i.e., 0 1 2 4 5 6 7 might become 4 5 6 7 0 1 2).
# 
# You are given a target value to search. If found in the array return its 
# index, otherwise return -1.
# 
# You may assume no duplicate exists in the array.
# ==============================================================================
# Method: brute force; python build-in
# Time Complexity: O(n)
# Space Complexity: O(1)
# ==============================================================================

class Solution:
    # @param {integer[]} nums
    # @param {integer} target
    # @return {integer}
    def search(self, nums, target):
        return nums.index(target) if target in nums else -1

File: 081-Search-in-Rotated-Sorted-Array-II/CPP/Solution02.cpp
/*
==============================================================================
Author:   Tao Li (taoli@ucsd.edu)
Date:     May 7, 2015
Question: 081-Search-in-Rotated-Sorted-Array-II
Link:     https://leetcode.com/problems/search-in-rotated-sorted-array-ii/
==============================================================================
Follow up for "Search in Rotated Sorted Array":
What if duplicates are allowed?

Would this affect the run-time complexity? How and why?

Write a function to determine if a given target is in the array.
==============================================================================
Method: loop; brute force
Time Complexity: O(n)
Space Complexity: O(1)
==============================================================================
*/

#include <vector>

using namespace std;

class Solution {
public:
    bool search(vector<int>& nums, int target) {
        for (int i = 0; i < nums.size(); i ++) {
            if (nums[i] == target) return true;
        }
        return false;
    }
};

/* Unit Test */
int main(int argc, char const *argv[])
{
    /* code */
    return 0;
}